src/librustdoc/config.rs

```diff
@@ -496,6 +498,21 @@ impl Options {
         let enable_per_target_ignores = matches.opt_present("enable-per-target-ignores");
         let document_private = matches.opt_present("document-private-items");
         let document_hidden = matches.opt_present("document-hidden-items");
+        let source_code_external_url = matches.opt_str("source-code-external-url").map(|mut h| {
+            if !h.ends_with('/') {
+                h.push('/');
+            }
+            h
+        });
+        if let Some(ref source_code_external_url) = source_code_external_url {
+            if !source_code_external_url.starts_with("http://")
+                && !source_code_external_url.starts_with("https://")
+            {
+                diag.struct_err("option `--source-code-external-url` argument must be an URL")
+                    .emit();
+                return Err(1);
+            }
+        }
```

Comment:
I know we don't tend to like adding dependencies but surely there's a good reason here to use a proper URL handling library for url joining, etc?

Reply:
If we don't care either or not it's a full URL, then I guess it can even be removed.

Reply:
Marking as resolved since this code was removed.